### PR TITLE
docs: improve LSP8 Natspec comments + add auto-generate docs

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

## 📄 Documentation

Improve Natspec comments for `LSP8IdentifiableDigitalAsset` and add auto-generated docs.

### CODE REVIEW

File: contracts/LSP7DigitalAsset/LSP7DigitalAssetCore.sol
Comment (b00ste) [inline contracts/LSP7DigitalAsset/LSP7DigitalAssetCore.sol:1]: This is not LSP8, I guess you mistakingly included it

File: contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol [hunk 1]
```diff
@@ -46,10 +45,10 @@ interface ILSP8IdentifiableDigitalAsset is IERC165, IERC725Y {
     );
 
     /**
-     * @dev Emitted when `tokenOwner` disables `operator` for `tokenId`.
-     * @param operator The address revoked from operating
-     * @param tokenOwner The token owner
-     * @param tokenId The tokenId `operator` is revoked from operating
+     * @dev Emitted when `tokenOwner` disables `operator` to transfer or burn `tokenId` on its behalf.
```
Comment (CallumGrindle) [inline contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol:48]:
```suggestion
     * @dev Emitted when `tokenOwner` disables `operator` to transfer or burn `tokenId.
```

File: contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol [hunk 2]
```diff
@@ -46,10 +45,10 @@ interface ILSP8IdentifiableDigitalAsset is IERC165, IERC725Y {
...
+     * @param operator The address revoked as an operator.
```
Comment (CallumGrindle) [inline contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol:49]: maybe smth like

```suggestion
     * @param operator The address revoked from the operator array.
```

File: contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol [hunk 3]
```diff
@@ -99,63 +97,64 @@ interface ILSP8IdentifiableDigitalAsset is IERC165, IERC725Y {
...
      *
+     * @custom:requirements
      * - `tokenId` must exist.
-     * - caller must be current `tokenOwner` of `tokenId`.
+     * - caller must be the {tokenOwnerOf} `tokenId`.
+     * - the owner of a `tokenId` cannot grant revoke itself as an `operator` (`operator` cannot be the calling address).
      * - `operator` cannot be the zero address.
      *
-     * Emits a {RevokedOperator} event.
+     * @custom:events {RevokedOperator} event event with address of the operator being revoked for the caller (token owner)..
```
Comment (CallumGrindle) [inline contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol:128]:
```suggestion
     * @custom:events {RevokedOperator} event with address of the operator being revoked for the caller (token owner)..
```

File: contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol [hunk 4]
```diff
@@ -15,14 +15,13 @@ interface ILSP8IdentifiableDigitalAsset is IERC165, IERC725Y {
...
-     * @param tokenId The tokenId transferred
-     * @param allowNonLSP1Recipient When set to TRUE, `to` may be any address but
-     * when set to FALSE `to` must be a contract that supports LSP1 UniversalReceiver
-     * @param data Additional data the caller wants included in the emitted event, and sent in the hooks to `from` and `to` addresses
+     * @dev Emitted when `tokenId` token is transferred from the `from` to the `to` address.
+     * @param operator The address of operator that sent the `tokenId`
+     * @param from The previous owner of the `tokenId`
+     * @param to The new owner of `tokenId`
+     * @param tokenId The tokenId that was transferred
+     * @param allowNonLSP1Recipient If the token transfer enforces the `to` recipient address to be a contract that implements the LSP1 standard or not.
```
Comment (CallumGrindle) [inline contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol:23]: did this replace force? we should update the standard doc if so

Comment (CallumGrindle) [reply contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol:23]: Also I think maybe `forceNonLSP1Recipient` is a better name? So its a bit more clear setting to true is risky

File: contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.sol [hunk 5]
```diff
@@ -17,9 +17,15 @@ import {
...
 
 /**
- * @title LSP8IdentifiableDigitalAsset contract
+ * @title Implementation of a LSP8 Identifiable Digital Asset, a contract that represents a non-fungible token.
  * @author Matthew Stevens
- * @dev Implementation of a LSP8 compliant contract.
+ *
+ * @dev Standard implementation contract of the LSP8 standard.
+ *
+ * Minting and transferring are done using by giving a unique `tokenId`.
```
Comment (CallumGrindle) [inline contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.sol:25]:
```suggestion
 * Minting and transferring are done by providing a unique `tokenId`.
```

File: contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol (hunk 4 above)
Comment (CJ42) [reply contracts/LSP8IdentifiableDigitalAsset/ILSP8IdentifiableDigitalAsset.sol:23]: Yes this changed to `allowNonLSP1Recipient`.
Really like this newly suggested name 👌

**Review by YamenMerhi — APPROVED** [review]

**Pull request merged**